### ISSUE TITLE
Limit amount of products that can be ordered according to stock levels.

### ISSUE THREAD

**Issue by Risha91** [vimes1984/woocommerce_variable_stock_management#6]
Let's say that i want to sell olive oil in my woocommerce shop

So i set up a variable product with stock of 5000 for main product and with variable options for 100ml, 500ml and 1000ml which can be ordered.

The problem appears when user decides to buy 10 products of 1000ml olive oil, which is in total 10000ml. The user is still able to order the product, even if the stock is set to 5000ml.

Is there a way to prevent that?

**Comment by vimes1984** [comment]
if they haven't checked out yet then yes they'd still be able to purchase
it since it's first come first served whoever chekcs out first gets the
prize...
but thats the same with all of woo

**Comment by Risha91** [comment]
Im not sure you understand my problems.

Your plugin is great for deducting stock of the main product based on its variation. So if someone buys 10 same items of a 1000ml Olive oil variation, you plugin will properly deduct its stock by 10000ml.

But is there a way it wouldnt allow the user to buy 10.000ml if the stock is set to 5000ml at the main product level.

**Comment by vimes1984** [comment]
ohhh yeah soft validate the amount of stock availalble against what the
user is trying to add to cart, but if I remeber correctly the plugin
allready does that. It shouldn't allow you to over order..
I built it for a client and I have it working with all sorts of custom
front end validation...

**Comment by Risha91** [comment]
Sadly it does not that for me. It allows the buyer to over order any product, even if i have "Dont allow backorders" checked on each product. So my stock goes from 50 to -70 sometimes.

Can i hope that it will be looked at in new plugin version? Im really dependant on your plugin sice most of my products are sold in variations of different units of measurement.

Thanks for all the good job!

**Comment by vimes1984** [comment]
https://github.com/vimes1984/woocommerce_variable_stock_management/blob/master/includes/class-catch-add-to-cart.php

line 87 is what does the validation for the add to cart which shouldn't
allow you to place in cart whats not available..
If you find a bug please feel free to patch the function and submit a pull
request I'll gladly pull in the code !

Chris

**Comment by Risha91** [comment]
Yeah as i see you are checking and validating against the stock of the main product which is alright, however each variation should be checked for the same value of its deduction.

Example: Main stock is set to 100
Variation 1: 1ml set deduct stock by 1
Variation 2: 5ml set deduct stock by 5
Variation 3: 10ml set deduct stock by 10

All three variations can currently be bought in quantity of 100 (which is the stock of main product), which counts to as following: variation 1 = 100, variation 2 = 500 and variation 3 = 1000.

Each variation should not be over ordered above the stock of the main product.

I hope you see the problem, since im far from being able to code it myself sadly. Im willing to pay for the pro version if such would come out including the support for the above problem.

Cheers.

**Comment by vimes1984** [comment]
send me a url

**Comment by stonejam** [comment]
Hi - any update with this issue? I am running into the same problem. I have a single, 2 pack, and 3 pack variation of the same product. I'd like to set a "master" inventory stock level (done by setting the stock quantity at the product level, rather than at the variation level). However, when I set the stock level to 10, I am able to add 10 to singles as well as 10 "2 packs" or 10 "3 packs" to my cart, rather than 10, 5, or 3 respectively. Back orders are set to "not allow". Any idea how to fix this?

**Comment by vimes1984** [comment]
Ummm I dont' support this persay it was just a quick plugin to solve a clients issue, but feel free to patch the code and I'd be happy to pull in the new version. 
Kind regards
Chris

**Comment by stonejam** [comment]
Thanks for the quick response! The plug-in works great otherwise and really is vital to efficient and accurate inventory management. However, without the properly constrained "add to cart" options, the potential for stock outs and unhappy customers is pretty high. Unfortunately, I can't write the code myself... don't know how. Otherwise I would be happy to help. Is there any chance you would be willing to attack the issue? Or perhaps know of anyone in the coding community who would be able to do that? Thank you so much for your contributions, thus far!

Warm regards, 
J

**Comment by vimes1984** [comment]
I'd be happy to attack it but I have paying jobs at the moment that are taking up my time.... 
You can alwasy try codeable.io for your customization needs though :dancer:

**Comment by stonejam** [comment]
oh wow. Great resource. Thanks for all your help!

**Comment by vimes1984** [comment]
I'm also on codeable if you need me to take a look at it :D

**Comment by stonejam** [comment]
How much would you charge for a project like this?

**Comment by vimes1984** [comment]
couple of hours I should think..

**Comment by stonejam** [comment]
How much do charge per hour?

couple of hours I should think..

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub
<https://github.com/vimes1984/woocommerce_variable_stock_management/issues/6#issuecomment-209901385>

**Comment by vimes1984** [comment]
80USD

**Comment by stonejam** [comment]
Hi, Chris! Sorry that I took a while to get back to you. We would love to
contract you to make that change for us! Can you give me some more info on
the process? This is something we would love to have resolved as soon as
possible.

Thanks so much,
J @ Cheerful Vet

80USD

—
You are receiving this because you commented.
Reply to this email directly or view it on GitHub
<https://github.com/vimes1984/woocommerce_variable_stock_management/issues/6#issuecomment-210873490>

**Comment by stonejam** [comment]
Hi again, Chris -

I haven't heard back from you so I thought I'd check in. I'm really hoping
to get this code written as soon as possible. Please contact me with more
details.

Thanks again for all your help,
J

**Comment by vimes1984** [comment]
I'd completly forgotten, I've just started a new job.

**Comment by stonejam** [comment]
Oh, does that mean you're not able to do this? Or could you work on this in
between working on your other projects?

**Comment by vimes1984** [comment]
I don't have time to look at it... 
I'm sorry but I can find you someone on: codable.io ( who I've taken the job with :D ) to lend  a hand though... 
chris
